### PR TITLE
Format error message for ops

### PR DESCRIPTION
**Pull request by kuke**
crf_decoding
edit_distance
expand_as
linear_chain_crf
sequence_conv
sequence_enumerate
sequence_expand
sequence_pool

### CODE REVIEW

File: paddle/fluid/operators/crf_decoding_op.cc [hunk 1]
```diff
@@ -89,41 +89,57 @@ class CRFDecodingOp : public framework::OperatorWithKernel {
...
-    PADDLE_ENFORCE_EQ(ctx->HasInput("Transition"), true,
-                      "Input(Transition) should be not null.");
-
-    PADDLE_ENFORCE_EQ(ctx->HasOutput("ViterbiPath"), true,
-                      "Output(ViterbiPath) should be not null.");
+    OP_INOUT_CHECK(ctx->HasInput("Emission"), "Input", "Emission",
+                   "CRFDecoding");
+    OP_INOUT_CHECK(ctx->HasInput("Transition"), "Input", "Transition",
+                   "CRFDecoding");
+    OP_INOUT_CHECK(ctx->HasOutput("ViterbiPath"), "Input", "ViterbiPath",
```
Comment (lfchener) [inline paddle/fluid/operators/crf_decoding_op.cc]: 第二个参数应该是 'Output' ?

File: paddle/fluid/operators/sequence_ops/sequence_expand_op.cc [hunk 2]
```diff
@@ -194,9 +211,10 @@ class SequenceExpandOpGrad : public framework::OperatorWithKernel {
 
  protected:
   void InferShape(framework::InferShapeContext* ctx) const override {
-    PADDLE_ENFORCE(ctx->HasInput("X"), "Input(X) should not be null.");
-    PADDLE_ENFORCE(ctx->HasInput(framework::GradVarName("Out")),
-                   "Input(Out@GRAD) should not be null.");
+    OP_INOUT_CHECK(ctx->HasInput("X"), "Input", "X", "SequenceExpandOpGrad");
+    OP_INOUT_CHECK(ctx->HasInput(framework::GradVarName("Out")), "Input",
+                   ctx->HasInput(framework::GradVarName("Out")),
```
Comment (lfchener) [inline paddle/fluid/operators/sequence_ops/sequence_expand_op.cc]: ctx->HasInput(framework::GradVarName("Out")) -> framework::GradVarName("Out")

File: paddle/fluid/operators/sequence_ops/sequence_pool_op.cc [hunk 3]
```diff
@@ -113,16 +112,26 @@ class SequencePoolGradOp : public framework::OperatorWithKernel {
...
+                          og_dims.size(), og_dims));
     for (int64_t i = 1; i < og_dims.size(); ++i) {
-      PADDLE_ENFORCE_EQ(og_dims[i], x_dims[i], "The dimension mismatch.");
+      PADDLE_ENFORCE_EQ(
+          og_dims[i], x_dims[i],
+          platform::errors::InvalidArgument(
+              "The dimension mismatch between Input(OUT@GRAD) and "
+              "Input(X). Received Input(OUT@GRAD): input rank %u, "
+              "input shape [%s]; received Input(X): input rank %u, "
+              "input shape [%u].",
```
Comment (lfchener) [inline paddle/fluid/operators/sequence_ops/sequence_pool_op.cc]: %u -> %s

File: paddle/fluid/operators/crf_decoding_op.cc [hunk 4]
```diff
@@ -132,20 +148,32 @@ class CRFDecodingOp : public framework::OperatorWithKernel {
...
       if (ctx->IsRuntime() || (emission_dims[0] > 0 && label_dims[0] > 0)) {
         PADDLE_ENFORCE_EQ(
             emission_dims[0], label_dims[0],
-            "The first dimension of Input(Emission) and Input(Label) "
-            "should be the same.");
+            platform::errors::InvalidArgument(
+                "The first dimension of Input(Emission) and Input(Label) "
+                "should be the same. But received Input(Emission): rank %u, "
+                "shape "
+                "[%u]; received Input(Label): rank %u, shape [%s].",
```
Comment (iclementine) [inline paddle/fluid/operators/crf_decoding_op.cc]: should shape be formated as str(%s)?

File: paddle/fluid/operators/linear_chain_crf_op.cc [hunk 5]
```diff
@@ -168,49 +171,88 @@ class LinearChainCRFOp : public framework::OperatorWithKernel {
...
+              label_dims.size(), label_dims));
       if (ctx->IsRuntime()) {
         PADDLE_ENFORCE_EQ(emission_dims[0], label_dims[0],
-                          "The batch size of Input(Emission) and Input(Label) "
-                          "should be the same.");
+                          platform::errors::InvalidArgument(
+                              "The batch size of Input(Emission) "
+                              "and Input(Label) should be the same. But "
+                              "received Input(Emission): "
+                              "rank %u, shape [%u]; received Input(Label): "
```
Comment (iclementine) [inline paddle/fluid/operators/linear_chain_crf_op.cc]: shape should be formatted as str?

File: paddle/fluid/operators/linear_chain_crf_op.cc [hunk 6]
```diff
@@ -168,49 +171,88 @@ class LinearChainCRFOp : public framework::OperatorWithKernel {
...
+                              emission_dims.size(), emission_dims,
+                              label_dims.size(), label_dims));
         PADDLE_ENFORCE_EQ(emission_dims[1], label_dims[1],
-                          "The max length of Input(Emission) and Input(Label) "
-                          "should be the same.");
+                          platform::errors::InvalidArgument(
+                              "The max length of Input(Emission) "
+                              "and Input(Label) should be the same. But "
+                              "received Input(Emission): "
+                              "rank %u, shape [%u]; received Input(Label): "
```
Comment (iclementine) [inline paddle/fluid/operators/linear_chain_crf_op.cc]: shape should be formated as str?

File: paddle/fluid/operators/linear_chain_crf_op.cc [hunk 7]
```diff
@@ -168,49 +171,88 @@ class LinearChainCRFOp : public framework::OperatorWithKernel {
...
       if (ctx->IsRuntime()) {
         PADDLE_ENFORCE_EQ(
             emission_dims[0], label_dims[0],
-            "The height of Input(Emission) and the height of Input(Label) "
-            "should be the same.");
+            platform::errors::InvalidArgument(
+                "The first dimension of Input(Emission) and Input(Label) "
+                "should be the same. But received Input(Emission): rank %u, "
+                "shape "
+                "[%u]; received Input(Label): rank %u, shape [%s].",
```
Comment (iclementine) [inline paddle/fluid/operators/linear_chain_crf_op.cc]: shape should be formatted as str?

File: paddle/fluid/operators/sequence_ops/sequence_expand_op.cc [hunk 8]
```diff
@@ -50,31 +51,47 @@ class SequenceExpandOp : public framework::OperatorWithKernel {
...
-                       "size of Input(Y)'s referred level lod.");
+        PADDLE_ENFORCE_EQ(
+            x_lod[0].size(), y_lod[ref_level].size(),
+            platform::errors::InvalidArgument(
+                "Level number of Input(X)'s lod could be 0. Otherwise "
+                "size of Input(X)'s first level lod should be equal to "
+                "size of Input(Y)'s referred level lod. But received: "
+                "Input(X).lod[0].size() = %u, Input(Y).lod[%d].size() = "
+                "%u",
+                x_lod[0].size(), y_lod[ref_level].size()));
```
Comment (iclementine) [inline paddle/fluid/operators/sequence_ops/sequence_expand_op.cc]: there are only 2 values, which is not enough for % holders?

**Review by iclementine — COMMENTED** [review]
some typos stated

File: paddle/fluid/operators/crf_decoding_op.cc (hunk 1 above)
Comment (kuke) [reply paddle/fluid/operators/crf_decoding_op.cc]: Done

File: paddle/fluid/operators/sequence_ops/sequence_expand_op.cc (hunk 2 above)
Comment (kuke) [reply paddle/fluid/operators/sequence_ops/sequence_expand_op.cc]: Done

File: paddle/fluid/operators/sequence_ops/sequence_pool_op.cc (hunk 3 above)
Comment (kuke) [reply paddle/fluid/operators/sequence_ops/sequence_pool_op.cc]: Done

File: paddle/fluid/operators/crf_decoding_op.cc (hunk 4 above)
Comment (kuke) [reply paddle/fluid/operators/crf_decoding_op.cc]: Done

File: paddle/fluid/operators/linear_chain_crf_op.cc (hunk 5 above)
Comment (kuke) [reply paddle/fluid/operators/linear_chain_crf_op.cc]: Done

File: paddle/fluid/operators/sequence_ops/sequence_expand_op.cc (hunk 8 above)
Comment (kuke) [reply paddle/fluid/operators/sequence_ops/sequence_expand_op.cc]: Done

File: paddle/fluid/operators/linear_chain_crf_op.cc (hunk 7 above)
Comment (kuke) [reply paddle/fluid/operators/linear_chain_crf_op.cc]: Done

File: paddle/fluid/operators/linear_chain_crf_op.cc (hunk 6 above)
Comment (kuke) [reply paddle/fluid/operators/linear_chain_crf_op.cc]: Done

File: paddle/fluid/operators/crf_decoding_op.cc [hunk 9]
```diff
@@ -89,41 +89,57 @@ class CRFDecodingOp : public framework::OperatorWithKernel {
...
+    OP_INOUT_CHECK(ctx->HasOutput("ViterbiPath"), "Output", "ViterbiPath",
+                   "CRFDecoding");
 
     auto emission_dims = ctx->GetInputDim("Emission");
     bool has_length = ctx->HasInput("Length");
 
     if (has_length) {
       PADDLE_ENFORCE_EQ(emission_dims.size(), 3,
-                        "The Input(Emission) should be a 3-D tensor.");
+                        platform::errors::InvalidArgument(
```
Comment (XiaoguangHu01) [inline paddle/fluid/operators/crf_decoding_op.cc:104]: 指明一下出错的OP

File: paddle/fluid/operators/crf_decoding_op.cc [hunk 10]
```diff
@@ -89,41 +89,57 @@ class CRFDecodingOp : public framework::OperatorWithKernel {
...
+                            emission_dims.size(), emission_dims));
     } else {
       PADDLE_ENFORCE_EQ(emission_dims.size(), 2,
-                        "The Input(Emission) should be a 2-D tensor.");
+                        platform::errors::InvalidArgument(
+                            "The Input(Emission) should be a 2-D tensor. But "
+                            "received: input rank %u, input shape [%s].",
+                            emission_dims.size(), emission_dims));
     }
-    PADDLE_ENFORCE_NE(emission_dims[0], 0,
```
Comment (XiaoguangHu01) [inline paddle/fluid/operators/crf_decoding_op.cc:110]: 这个检查删除的原因？

File: paddle/fluid/operators/sequence_ops/sequence_conv_op.h [hunk 11]
```diff
@@ -90,8 +95,12 @@ class SequenceConvGradKernel : public framework::OpKernel<T> {
     int context_stride = context.Attr<int>("contextStride");
     bool padding_trainable = context.Attr<bool>("paddingTrainable");
 
-    PADDLE_ENFORCE_EQ(in->lod().size(), 1UL,
-                      "Only support one level sequence now.");
+    PADDLE_ENFORCE_EQ(
+        in->lod().size(), 1UL,
+        platform::errors::InvalidArgument(
+            "Only support input sequence with lod level equal to 1 at "
```
Comment (XiaoguangHu01) [inline paddle/fluid/operators/sequence_ops/sequence_conv_op.h:101]: 指明下出错的OP名和参数名

File: paddle/fluid/operators/sequence_ops/sequence_conv_op.h [hunk 12]
```diff
@@ -41,9 +41,14 @@ class SequenceConvKernel : public framework::OpKernel<T> {
...
         in->lod().empty(), false,
-        "Input(X) Tensor of SequenceConvOp does not contain LoD information.");
-    PADDLE_ENFORCE_EQ(in->lod().size(), 1UL,
-                      "Only support one level sequence now.");
+        platform::errors::InvalidArgument("Input(X) Tensor of SequenceConvOp "
+                                          "does not contain LoD information."));
+    PADDLE_ENFORCE_EQ(
+        in->lod().size(), 1UL,
+        platform::errors::InvalidArgument(
+            "Only support input sequence with lod level equal to 1 at "
```
Comment (XiaoguangHu01) [inline paddle/fluid/operators/sequence_ops/sequence_conv_op.h:49]: 指明下出错的OP名和参数名

File: paddle/fluid/operators/sequence_ops/sequence_expand_op.cc [hunk 13]
```diff
@@ -50,31 +51,47 @@ class SequenceExpandOp : public framework::OperatorWithKernel {
...
+              "received: lod level %u.",
+              y_lod.size()));
+      PADDLE_ENFORCE_EQ(
           ref_level == -1 ||
               (ref_level >= 0 && ref_level < static_cast<int>(y_lod.size())),
-          "Invlid `ref_level`, which should be either equal to -1 "
-          "or in [0, %d)",
-          y_lod.size());
+          true, platform::errors::InvalidArgument(
+                    "Invlid `ref_level`, which should be either equal to -1 "
```
Comment (XiaoguangHu01) [inline paddle/fluid/operators/sequence_ops/sequence_expand_op.cc:68]: 确认下ref_level，对于开发者来说是否可以理解

**Review by XiaoguangHu01 — CHANGES_REQUESTED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]

**Review by Boyan-Liu — APPROVED** [review]

**Pull request merged**